web3/utils/datastructures.py

```diff
@@ -168,3 +173,18 @@ def __reversed__(self):
         if not isinstance(elements, Sequence):
             elements = list(elements)
         return iter(elements)
+
+
+class HexBytes(bytes):
+    def __new__(cls, val):
+        bytesval = hexstr_if_str(to_bytes, val)
+        return super().__new__(cls, bytesval)
+
+    def __init__(self, val):
+        super().__init__()
```

Comment:
*curious*

This *looks* like a passthrough except that it doesn't actually pass `val` through.  Since tests are passing, I suspect this actually works, but I'm curious why (which indicates this could maybe use a small comment explaining why `val` isn't passed through).

Reply:
Yeah, that was me trying to figure out how to subclass `bytes` and forgetting to remove the cruft. I just removed it.

X) I really like this one: `BytesThatLookLikeHexidecimalButThatYouCanTreatLikeBytes`